tests/e2e/lib.sh

```diff
@@ -43,42 +43,71 @@ deploy_stackrox() {
-    if [ ! -f "$password_file" ]; then
-        die "password file $password_file not found after deploying central"
-    fi
-    kubectl -n stackrox exec deploy/central -- roxctl --insecure-skip-tls-verify \
-        --password "$(cat "$password_file")" \
-      central init-bundles generate stackrox-init-bundle --output - 1> stackrox-init-bundle.yaml
+    # Repo name can't be too long or `helm search repo [REPO_NAME] -l` cuts off part of the name and the regex below fails.
+    helm_repo_name="tmp-srox-compat"
+    helm repo add "${helm_repo_name}" https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource
+    helm repo update
 
-    deploy_sensor_from_helm_charts "$target_version" ./stackrox-init-bundle.yaml
+    latest_tag=$(make tag)
 
-    echo "Sensor deployed. Waiting for sensor to be up"
-    sensor_wait
+    helm_charts=$(helm search repo ${helm_repo_name} -l)
+    central_regex="${helm_repo_name}/stackrox-central-services[ \t]*.${CENTRAL_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
+    sensor_regex="${helm_repo_name}/stackrox-secured-cluster-services[ \t]*.${SENSOR_CHART_VERSION_OVERRIDE}[ \t]*.([0-9]+\.[0-9]+\.[0-9]+)"
 
-    # Bounce collectors to avoid restarts on initial module pull
-    kubectl -n stackrox delete pod -l app=collector --grace-period=0
+    # If the central version is the latest the default behavior of deploy_central() is correct for compatibility tests
+    if  [[ $helm_charts =~ $central_regex ]]; then
+        ci_export CENTRAL_CHART_DIR_OVERRIDE "./central-chart-dir-compatibility-tests"
+        central_chart="${helm_repo_name}/stackrox-central-services"
+        helm pull "${central_chart}" --version "${CENTRAL_CHART_VERSION_OVERRIDE}" --untar --untardir "${CENTRAL_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" != "$CENTRAL_CHART_VERSION_OVERRIDE" ]]; then
+        echo >&2 "stackrox-central-services helm chart for version ${CENTRAL_CHART_VERSION_OVERRIDE} not found in ${helm_repo_name} repo nor is it the latest tag."
+        exit 1
+    fi
 
-    sensor_wait
+    # If the sensor version is the latest the default behavior of deploy_sensor() is incorrect, because it will deploy
+    # a sensor version to match the central version. In our tests we want to test latest sensor vs older central too,
+    # and since latest sensor is not available in the repo either the chart is created here in the elif case.
+    if [[ $helm_charts =~ $sensor_regex ]]; then
+        sensor_chart="${helm_repo_name}/stackrox-secured-cluster-services"
+        ci_export SENSOR_CHART_DIR_OVERRIDE "./sensor-chart-dir-compatibility-tests"
+        helm pull "${sensor_chart}" --version "${SENSOR_CHART_VERSION_OVERRIDE}" --untar --untardir "${SENSOR_CHART_DIR_OVERRIDE}"
+    elif [[ "$latest_tag" == "$SENSOR_CHART_VERSION_OVERRIDE" ]]; then
+        if [[ $(roxctl version) != "$latest_tag" ]]; then
+            echo >&2 "Roxctl version $(roxctl version) is not equal to latest tag ${latest_tag}"
```

Comment:
```suggestion
            >&2 echo "Reported roxctl version $(roxctl version) is different from requested tag ${latest_tag}. It won't be possible to get helm charts for ${latest_tag}. Please check test setup."
```
... or something similar. Basically, I suggest to mention practical consequences of tag mismatch and have some call-to-action to suggest engineer where to start troubleshooting.

Reply:
Now that I think of it, should I prepend `ERROR:` or `FAIL:` to make that message easier to find in the log?

Reply:
Up to you. OSCI highlights `stderr` with a different color than `stdout` so in my view it's good enough already.